src/Workspaces/Core/Portable/Workspace/Host/DocumentService/DocumentPropertiesService.cs

```diff
@@ -16,5 +18,11 @@ internal class DocumentPropertiesService : IDocumentService
         /// but is not passed to the compiler when the containing project is built.
         /// </summary>
         public virtual bool DesignTimeOnly => false;
+
+        /// <summary>
+        /// The LSP client name that should get the diagnostics produced by this document; any other source
+        /// will not show these diagnostics. If null, the diagnostics do not have this special handling.
+        /// </summary>
+        public virtual string? DiagnosticsLspClientName => "hello";
```

Comment:
lol

Reply:
I, uh, thought that was removed. :smile:

Reply:
(You discovered my secret debugging technique!)